app/src/models/uncommitted-changes-strategy.ts

```diff
@@ -20,3 +24,25 @@ export const askToStash: UncommittedChangesStrategy = {
 export const stashOnCurrentBranch: UncommittedChangesStrategy = {
   kind: UncommittedChangesStrategyKind.StashOnCurrentBranch,
 }
+export const moveToNewBranch: UncommittedChangesStrategy = {
+  kind: UncommittedChangesStrategyKind.MoveToNewBranch,
+  transientStashEntry: null,
+}
+
+export function getUncommittedChangesStrategy(
+  kind: UncommittedChangesStrategyKind
+) {
+  switch (kind) {
+    case UncommittedChangesStrategyKind.AskForConfirmation:
+      return askToStash
+    case UncommittedChangesStrategyKind.MoveToNewBranch:
+      return moveToNewBranch
+    case UncommittedChangesStrategyKind.StashOnCurrentBranch:
+      return stashOnCurrentBranch
+    default:
+      return assertNever(
+        kind,
+        `Unknown UncommittedChangesStrategyKind: ${kind}`
+      )
+  }
+}
```

Comment:
I have some concerns about this function, but I think the core of my concern is that it took me a while to figure out what it was doing. Perhaps documenting this to clarify that its primary purpose is to map a kind to an actual `UncommittedChangesStrategy`?

It feels to me like there's a simpler version of this overall approach, but I'm not sure what it is yet...Is the main reason this exists so that we can store an `UncommittedChangesStrategyKind` in `AppState` and convert it to an `UncommittedChangesStrategy` whenever we use it? perhaps we could store the result of this function in `AppState` and avoid doing this conversion in many places?